### PR TITLE
Newsletter 1.0 for Apache project

### PR DESCRIPTION
**Pull request by chibenwa**
View the visuals of the newsletter online : https://github.com/chibenwa/james-project/blob/newsletter-1/newletter.md

### CODE REVIEW

File: newletter.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+We now use James as the mail server behind OpenPaaS. Thus we deployed it and use it on an every-day base.
+
+Our deployment is based on the following components :
+ - We enabled IMAP, SMTP and JMAP protocols
+ - We use Cassandra for storing e-mails
+ - We rely on ElasticSearch for searching e-mails
+ - We authenticate our users with LDAP
+ - We use Guice for bringing all pieces together.
+ - We plan to bring soon Sieve and ManageSieve part of this deployment as many peaple request filtering.
```
Comment (quynhn) [inline newletter.md]: s/peaple/?

File: newletter.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Let me share a few numbers with you :
+
+ - We handle about 100 users
+ - We have so far a number of 1.500.000 e-mails, that we imported using IMAP-sync scripts
+ - We receive arounf 15.000 incoming e-mails a day
```
Comment (quynhn) [inline newletter.md]: s/arounf/?

File: newletter.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ - Then we added a naive SMTP implementation
+ - And finally contributed a IMAP DSL for Gatling
+ 
+ Running these load tests we :
+ 
+  - Succeeded to support 1.000 thunderbird like users
+  - However, it turned out we had problems with 10.000 users.
+ 
+ Additionnaly, we added metrics a bit everywhere in James using the brand new metrics API. We collect and export everything in ElasticSearch using Dropwizard metrics.
+ Then we graph it all using Grafana. This allow us to cellect all statistic and percentiles. We track so far :
```
Comment (quynhn) [inline newletter.md]: s/cellect/?

File: newletter.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+ 
+  - Protocols detailed time execution (and count)
+  - Percentile of mailet and matcher executions
+  - Enqueue and Dequeue time
+ 
+ All these solutions allow us to identify the componants that needs improvement. For instance receiving  too much incoming e-mails overload James with very heavy garbage collection.
```
Comment (quynhn) [inline newletter.md]: s/componants/?

File: newletter.md [hunk 5]
```diff
@@ -63,10 +63,26 @@ We are pretty happy with the current state of the James server. We will then pus
...
+  - We will continue doing bug fixes
+  - We need some additional performance enhancement, especially with IMAP SELECT command on large mailboxes
+  - We plan to start working again on the new website, which have been paused a few months
+
+## Usefull links
+
+ - Gatling: http://gatling.io/
+ - Gatling James (JMAP + SMTP): https://github.com/linagora/james-gatling
+ - Gatling IMAP: https://github.com/linagora/gatling-imap
+ - Dopwizar metrics: http://metrics.dropwizard.io/3.2.1/
```
Comment (quynhn) [inline newletter.md]: s/Dopwizar/?

File: newletter.md (hunk 1 above)
Comment (chibenwa) [inline newletter.md]: Corrected in latter commits

File: newletter.md (hunk 2 above)
Comment (chibenwa) [inline newletter.md]: Corrected in latter commits

File: newletter.md (hunk 3 above)
Comment (chibenwa) [inline newletter.md]: Corrected in latter commits

File: newletter.md (hunk 4 above)
Comment (chibenwa) [inline newletter.md]: corrected in later commits

File: newletter.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
+# Linagora's James newsletter
+
+[Linagora](http://linagora.com/) have a team of developpers devoted to the James server. We are already contributing for a few years to the project. This document will try to give you information about what we recently did for the James project, and give hints about the road-map we follow. We will also try to write it regulary.
```
Comment (rouazana) [inline newletter.md]: s/for the/in/ ?

File: newletter.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+## Tracking performance
+
+Everybody wants to read their e-mails fast. Thus the team made performance tracking a priority. And we developed the tools to follow performance.
+
+![Result of 1.000 thunderbird like users](gatling1.png)
+
+We started implementing [Gatling](http://gatling.io/) load testing for the James server :
+
+  - First with a [Gatling JMAP](https://github.com/linagora/james-gatling) implementation.
+  - Then we added a naive SMTP implementation
```
Comment (rouazana) [inline newletter.md]: Link?

File: newletter.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  - And finally contributed a [IMAP DSL for Gatling](https://github.com/linagora/gatling-imap)
```
Comment (rouazana) [inline newletter.md]: s/a/an/

File: newletter.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+![Result graph of 1.000 thunderbird like users](gatling2.png)
+
+Running these load tests we :
+
+  - Succeeded to support 1.000 thunderbird like users
+  - However, it turned out we had problems with 10.000 users.
```
Comment (rouazana) [inline newletter.md]: Please add the limitations of the server (only one server, with cassandra and ES on the same host)

File: newletter.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+Additionally, we added metrics a bit everywhere in James using the brand new metrics API. We collect and export everything in [ElasticSearch](https://github.com/elastic/elasticsearch-metrics-reporter-java) using [Dropwizard metrics](http://metrics.dropwizard.io/3.2.1/). Then we graph it all using [Grafana](http://grafana.com/). This allow us to collect all statistic and percentiles. We track so far :
+
+  - Protocols detailed time execution (and count)
+  - Percentile of mailet and matcher executions
+  - Enqueue and Dequeue time
```
Comment (rouazana) [inline newletter.md]: you can add a link to the committed graphs

File: newletter.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+  
+![James metrics and percentile for the JMAP protocol](grafana.png)
+
+All these solutions allow us to identify the components that needs improvement. For instance receiving too much incoming e-mails overload James with very heavy garbage collection. We then plan to move our mail queue to RabbitMQ, to parse e-mails only once on top of the mail pipeline, to reorganize a bit our configuration. A Camel upgrade (impossible in java-6) might also help.
+ 
+ ## MessageId refactoring
+ 
+ We succeeded to finish and merge the MessageId refactoring. This huge task allow us to address messages by their ID, outside of mailbox context.
+ 
+ This is required for JMAP protocol implementation.
```
Comment (rouazana) [inline newletter.md]: For the record that's not exactly required. You can have an implementation where each message belongs only to one folder, but then you cannot handle folders as labels.

File: newletter.md (hunk 9 above)
Comment (chibenwa) [inline newletter.md]: I'll also add CPU model, memory...

File: newletter.md (hunk 11 above)
Comment (chibenwa) [inline newletter.md]: Lol thanks for the info. Let's keep this hack for us.

**Comment by chibenwa** [comment]
Sent. I will keep the branch for image hosting, and for online reading

**Pull request closed without merging**